mock/api.js

```diff
@@ -1,381 +1,299 @@
-          title: 'test_measurement_type',
-          children: [
-            {
-              title: 'test_measurement_title',
-              children: [
-                {
-                  title: 'test_measurement_type-test_measurement_title-test_measurement_id',
-                  children: [
-                    {
-                      title: 'mean',
-                      dataIndex:
-                        'test_measurement_type-test_measurement_title-test_measurement_id-mean',
-                      key: 'test_measurement_type-test_measurement_title-test_measurement_id-mean',
-                    },
-                    {
-                      title: 'stddevpct',
-                      dataIndex:
-                        'test_measurement_type-test_measurement_title-test_measurement_id-stddevpct',
-                      key:
-                        'test_measurement_type-test_measurement_title-test_measurement_id-stddevpct',
-                    },
-                    {
-                      title: 'closest_sample',
-                      dataIndex:
-                        'test_measurement_type-test_measurement_title-test_measurement_id-closest_sample',
-                      key:
-                        'test_measurement_type-test_measurement_title-test_measurement_id-closest_sample',
-                    },
-                  ],
-                },
-              ],
-            },
-          ],
-        },
-      ],
-      run_name: 'test_run_name',
-      run_controller: 'test_controller',
-      iterations: {
-        test_iteration_1: {
-          name: 'test_iteration_1',
-          number: 1,
-          closest_sample: 1,
-          'test_measurement_type-test_measurement_title-test_measurement_id-closest_sample': 1,
-          'test_measurement_type-test_measurement_title-test_measurement_id-mean': 0.1,
-          'test_measurement_type-test_measurement_title-test_measurement_id-stddevpct': 1,
-          samples: {
-            'test_measurement_title-sample1': {
-              sample: {
-                closest_sample: 1,
-                mean: 0.1,
-                stddev: 0.1,
-                stddevpct: 1,
-                uid: 'test_measurement_id',
-                measurement_type: 'test_measurement_type',
-                measurement_idx: 0,
-                measurement_title: 'test_measurement_title',
-                '@idx': 0,
-                name: 'sample1',
-              },
-              benchmark: {
-                instances: 1,
-                max_stddevpct: 1,
-                message_size_bytes: 1,
-                primary_metric: 'test_measurement_title',
-                test_type: 'stream',
-              },
-              run: {
-                id: 'test_run_id',
-                controller: 'test_controller',
-                name: 'test_run_name',
-                script: 'test_script',
-                config: 'test_config',
-              },
+export const mockMappings = {
+  text_index: {
```

Comment:
We look for 
```
const index = endpoints.prefix + endpoints.run_index + indices[0];
```
in `search/fetchIndicexMapping` which according to the mock endpoints file becomes `test_prefix.v.run-data.2021-03` but the key given here is `test_index` which doesn't fetch mappings on to the search page.

Reply:
Thanks for catching this Aquib! This is an interesting edge case where we're unable to build the correct index due to api.js being referenced as part of the Node build process which means we don't have access to `window.endpoints`. I'm including a temporary fix that allows us to reference `test_index` when in the mock development environment. We'll want to work on adapting the server endpoint for fetching index mappings to take an index parameter so that the dashboard doesn't parse the response for a specific index name.